File: src/main/java/edu/jmi/openatom/server/openatomsystem/entity/RecruitmentCampaign.java
package edu.jmi.openatom.server.openatomsystem.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.sql.Timestamp;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@TableName("recruitment_campaign")
public class RecruitmentCampaign {
  @TableId(value = "id", type = IdType.AUTO)
  private Integer id;

  @TableField("club_id")
  private Integer clubId;

  private String name;

  @TableField("apply_start_at")
  private Timestamp applyStartAt;

  @TableField("apply_end_at")
  private Timestamp applyEndAt;

  @TableField("interview_start_at")
  private Timestamp interviewStartAt;

  @TableField("interview_end_at")
  private Timestamp interviewEndAt;

  @TableField("result_publish_at")
  private Timestamp resultPublishAt;

  @TableField("target_grades")
  private String targetGrades;

  @TableField("max_applicants")
  private Integer maxApplicants;

  @TableField("login_required")
  private Boolean loginRequired;

  @TableField("form_schema")
  private String formSchema;

  private String status;

  @TableField("created_at")
  private Timestamp createdAt;

  @TableField("updated_at")
  private Timestamp updatedAt;
}

File: frontend/ios_app/openatom/Models/RecruitmentCampaign.swift
import Foundation

struct ResponseRecruitmentDTO: Codable {
    let club: Club?
    let campaigns: [RecruitmentCampaign]?
    let departments: [ClubDepartment]?
}

struct ResponseRecruitmentDetailDTO: Codable {
    let club: Club?
    let campaign: RecruitmentCampaign?
    let departments: [ClubDepartment]?
}

struct RecruitmentCampaign: Codable, Identifiable {
    let id: Int
    let clubId: Int?
    let name: String?
    let applyStartAt: String?
    let applyEndAt: String?
    let interviewStartAt: String?
    let interviewEndAt: String?
    let resultPublishAt: String?
    let targetGrades: String?
    let maxApplicants: Int?
    let loginRequired: Bool?
    let formSchema: String?
    let status: String?
    let createdAt: String?
    let updatedAt: String?

    var parsedFormSchema: FormSchema? {
        guard let json = formSchema, let data = json.data(using: .utf8) else { return nil }
        return try? JSONDecoder().decode(FormSchema.self, from: data)
    }
}

struct FormSchema: Codable {
    let fields: [FormField]?
}

struct FormField: Codable, Identifiable {
    let key: String?
    let label: String?
    let type: String?
    let required: Bool?
    let options: [FormFieldOption]?
    let placeholder: String?

    var id: String { key ?? UUID().uuidString }
}

struct FormFieldOption: Codable, Identifiable {
    var id: String { value ?? UUID().uuidString }
    let label: String?
    let value: String?
}
